### ISSUE TITLE
400 User location is not supported for the API use.

### ISSUE THREAD

**Issue by YassineBouzid** [rokbenko/ai-playground#1]
400 User location is not supported for the API use.

**Comment by rokbenko** [comment]
Hi, @YassineBouzid.

I guess you're getting this error trying to run the script from the <a href="https://github.com/rokbenko/ai-playground/tree/main/gemini-tutorials/1-Gemini_Pro_API_starter">#1 Gemini Pro API starter</a> tutorial, right?

Unfortunately, the Gemini API is not available in any country. See the <a href="https://ai.google.dev/available_regions#available_regions">Available regions</a> to see if your country is listed there.

In fact, <a href="https://stackoverflow.com/questions/77659552/google-generative-ai-api-error-user-location-is-not-supported-for-the-api-use">I got an identical error</a> while I was preparing the code for the tutorial. **I simply used a VPN to connect to the US.** The error was gone.

Let me know if this solves your problem.

**Comment by YassineBouzid** [comment]
Hi, @rokbenko.
Thank you for your prompt response and the valuable workaround suggestion. I've tested the Gemini Pro API on my local machine, and it's working flawlessly even without a VPN.

However, the issue seems to be specific to the web host I'm using, where running the API is not allowed. I reached out to their support team to migrate my website to another cluster where the API is permitted, but unfortunately, they mentioned that it's not possible.

In their response, they suggested using a proxy server or VPN if it's allowed by Google. I'm currently exploring these options to ensure seamless API integration. If you have any additional insights or recommendations, I would greatly appreciate them.

Thanks again for your assistance.
